edweiss-app/__test__/components/lectures/remotecontrol/modal.test.tsx

```diff
@@ -0,0 +1,168 @@
+                </>
+            )),
+        },
+    };
+});
+
+// Authentication mock
+jest.mock('@/contexts/auth', () => ({
+    useAuth: jest.fn(),
+}));
+
+// BottomSheet to detect modal appearance
+jest.mock('@gorhom/bottom-sheet', () => ({
+    BottomSheetModal: jest.fn(({ children }) => (
+        <div>{children}</div>
+    )),
+    BottomSheetView: jest.fn(({ children }) => (
+        <div>{children}</div>
+    )),
+    BottomSheetBackdrop: jest.fn(),
+}));
+
+// Sign in fucntions
+jest.mock('@/config/firebase', () => ({
+    signInWithGoogle: jest.fn(),
+    signInAnonymously: jest.fn(),
+    callFunction: jest.fn(),
+}));
+
+
+// Firebase modules
+jest.mock('@react-native-firebase/auth', () => ({
+    currentUser: { uid: 'anonymous' },
+    signInAnonymously: jest.fn(() => Promise.resolve({ user: { uid: 'anonymous' } })),
+}));
+
+
+// Firebase functions
+jest.mock('@react-native-firebase/functions', () => {
+    return {
+        httpsCallable: jest.fn(() => jest.fn(() => Promise.resolve({ data: { status: true } })))
+    };
+});
+
+// Storage module
+jest.mock('@react-native-firebase/storage', () => {
+    return {
+        ref: jest.fn(() => ({
+            putFile: jest.fn(() => Promise.resolve({})),
+            getDownloadURL: jest.fn(() => Promise.resolve('mocked_download_url')),
+            delete: jest.fn(() => Promise.resolve({}))
+        })),
+    };
+});
+
+
+// Firestore communication with database
+jest.mock('@react-native-firebase/firestore', () => ({
+    __esModule: true,
+    default: jest.fn(() => Promise.resolve({})),
+}));
+
+
+
+// useDynamicDocs hooks
+jest.mock('@/hooks/firebase/firestore', () => ({
+    useDynamicDocs: jest.fn(() => [
+        { id: '1', data: () => ({ name: 'Sample Todo', status: 'yet' }) }
+    ])
+}));
+
+
+
+describe('LangSelectModal', () => {
+    it('renders all language buttons', () => {
+        const modalRef = React.createRef<BottomSheetModal>();
+        const { getByTestId } = render(
+            <LangSelectModal
+                modalRef={modalRef}
+                lang="english"
+                setLang={mockSetLang}
+                onClose={mockOnClose}
+            />
+        );
+
+
+        expect(getByTestId('lang-sel-close-button'));
+        availableLangs.forEach(lang => {
+            expect(getByTestId(`lang-but-${lang}`)).toBeTruthy();
+        });
+
+    });
+
+    it('calls setLang when a language button is pressed', () => {
+        const modalRef = React.createRef<BottomSheetModal>();
+        const { getByTestId } = render(
+            <LangSelectModal
+                modalRef={modalRef}
+                lang="english"
+                setLang={mockSetLang}
+                onClose={mockOnClose}
+            />
+        );
+
+        fireEvent.press(getByTestId('lang-but-french'));
+        expect(mockSetLang).toHaveBeenCalledWith('french');
+    });
+
+    it('calls setLang when a language button is pressed', () => {
```

Comment:
you could diferentiate the language in the test tittles since it seemed to be two repeated identical tests but in fact were diferent in the aspect of which language was being tested. I think just putting "calls setLang when French button is pressed" or English would be clearer.

Reply:
Oops, my bad 👀. I have copied the test code once I had the first one working 🫣.
You are entirely right, I will fix it right away. Good catch 🏌️.

Reply:
Happy to help